packages/cli/src/cms/commands/utils.ts

```diff
@@ -64,3 +67,62 @@ export function handleReuniteError(
 
   return exitWithError(errorMessage);
 }
+
+export function createCommandSunsetMiddleware(command: 'push' | 'push-status') {
+  const { middleware, buffer: warningBuffer } = createApiSunsetMiddleware({
+    bufferSunsets: true,
+  });
+
+  function sortWarningsBySunsetDate(warningBuffer: SunsetWarningsBuffer) {
+    // Sort warnings to report single nearest in future or already past warning
+    return warningBuffer.sort((a: SunsetWarning, b: SunsetWarning) => {
+      if (a.isSunsetExpired && b.isSunsetExpired) {
+        if (a.sunsetDate > b.sunsetDate) {
+          // b will shutdown earlier
+          return 1;
+        } else {
+          return -1;
+        }
+      }
+
+      if (a.isSunsetExpired) {
+        // a should come before b because it is already expired
+        return -1;
+      }
+
+      if (b.isSunsetExpired) {
+        // b should come before a because it is already expired
+        return 1;
+      }
+
+      if (a.sunsetDate > b.sunsetDate) {
+        // b will shutdown earlier
+        return 1;
+      } else {
+        return -1;
+      }
+    });
+  }
+
+  return {
+    middleware,
+    report: () => {
+      if (warningBuffer.length) {
+        const [{ isSunsetExpired, sunsetDate }] = sortWarningsBySunsetDate(warningBuffer);
+
+        const sunsetMessagePrevix = `Endpoint required for proper work of "${command}" command is`;
```

Comment:
Check variable spelling: probably `... Prefix`

This sentence doesn't make sense (I realise it's not a whole sentence). Something like:

```suggestion
        const sunsetMessagePrevix = `Support for "${command}" command is`;
```